dbt/include/snowflake/macros/adapters.sql

```diff
@@ -134,7 +135,10 @@
 
 {% endmacro %}
 
-{% macro snowflake__list_relations_without_caching(schema_relation, max_iter=10, max_results_per_iter=10000) %}
+{% macro snowflake__list_relations_without_caching(schema_relation) %}
+
+  {%- set max_results_per_iter = adapter.config.flags.get('list_relations_per_page', 10000) -%}
```

Comment:
Did we consult DX or PM folks on the names here? I'm just not sure if it's the most ergonomic.